### PR TITLE
Teuchos: Adding capability to broadcast & update parameterlists

### PR DESCRIPTION
**Pull request by csiefer2**
Auto-PR for SHA 4e67d63

### CODE REVIEW

**Comment by csiefer2** [comment]
@bartlettroscoe

File: packages/teuchos/comm/src/Teuchos_RawParameterListHelpers.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    if(overwrite) paramList->setParameters(*inParamList);
+    else paramList->setParametersNotAlreadySet(*inParamList);
+  }
+  else {
+    if (comm.getRank()==0) {
+      XMLParameterListWriter w;
+      std::string xmlString = toString(w.toXML(*inParamList));
+      int strsize = static_cast<int>(xmlString.size());
+      broadcast<int, int>(comm, 0, &strsize);
+      broadcast<int, char>(comm, 0, strsize, &xmlString[0]);
```
Comment (rppawlo) [inline packages/teuchos/comm/src/Teuchos_RawParameterListHelpers.cpp]: shouldn't the broadcasts be using the root variable?

Comment (bartlettroscoe) [reply packages/teuchos/comm/src/Teuchos_RawParameterListHelpers.cpp]: @rppawlo said:

> shouldn't the broadcasts be using the root variable?

Good catch.  I missed that (I was mostly looking at the unit tests).

My guess is that this was a copy and paste error from either:

* https://github.com/trilinos/Trilinos/blob/develop/packages/teuchos/comm/src/Teuchos_XMLParameterListHelpers.cpp#L58

or

* https://github.com/trilinos/Trilinos/blob/develop/packages/teuchos/comm/src/Teuchos_YamlParameterListHelpers.cpp#L60

Note that those functions do not take a `root` argument.  But in that context, they just read from a file from one process, so it does not really matter which process reads the file to broadcast.

Comment (csiefer2) [reply packages/teuchos/comm/src/Teuchos_RawParameterListHelpers.cpp]: Yup, cut and paste error.

**Comment by csiefer2** [comment]
@bartlettroscoe Requested changes made.  Please review again.

**Pull request merged**